vars/npmExecute.groovy

```diff
@@ -0,0 +1,65 @@
+import static com.sap.piper.Prerequisites.checkScript
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.Utils
+import groovy.transform.Field
+
+@Field def STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STEP_CONFIG_KEYS = [
+    /**
+     * Name of the docker image that should be used, in which node should be installed and configured. Default value is 'node:8-stretch'.
+     */
+    'dockerImage',
+    /**
+     * Which NPM command should be executed. Default value is 'run build'.
+     */
+    'npmCommand']
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS + [
+    /**
+     * Docker options to be set when starting the container.
+     */
+    'dockerOptions']
+/**
+ * Executes NPM commands inside a docker container.
+ * Docker image, docker options and npm commands can be specified or configured.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = checkScript(this, parameters) ?: this
+
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        new Utils().pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'scriptMissing',
+            stepParam1: parameters?.script == null
+        ], configuration)
+
+        try {
+            if (fileExists('package.json')) {
+                dockerExecute(script: script, dockerImage: configuration.dockerImage, dockerOptions: configuration.dockerOptions) {
+                    sh """
+                        npm install
+                        npm ${configuration.npmCommand}
```

Comment:
We implemented it a bit more flexible by running body() instead of a single commands. But there advantages and disadvantages for that approach.

Reply:
I added the closure before and [removed it](https://github.com/SAP/jenkins-library/pull/491/commits/cf2c2df6da98f0a51a5aff81c5657d289501096e), my intention was to keep Jenkinsfile simple and put the logic of npm project into package.json. Should we keep them both to keep compatibility? What is your opinion?

Reply:
Add body() with https://github.com/SAP/jenkins-library/pull/491/commits/9275e555588f24ec0e90d389307b5e25b3e08d6b